core/src/surplus.ts

```diff
@@ -54,11 +54,22 @@ const getAuctionState = async (network: string, earliestEndDate: Date, greatestB
     return 'just-started';
 };
 
+const _getContractAuctionDuration = async (contract: Contract) => {
```

Comment:
The contract is an object with functions so it doesn't cache well. Likely, this function will still be executed every time or wouldn't expire after switching the network. Instead, I suggest to use `network` as a parameter
```suggestion
const _getSurplusAuctionsDurationInSeconds = async (network: string) => {
```

Reply:
> The contract is an object with functions so it doesn't cache well.

do you have this claim by experience or is there a handy-dandy article i could read to understand why it is so?

Reply:
You probably know that in js there is a difference between primitive values (string, number, boolean, undefined, symbol, null, etc) and objects (objects, arrays, functions, classes, etc) in which `=` or `===` operators [work differently](https://flaviocopes.com/difference-primitive-types-objects/) and basically compare references instead of values.

I can't find what `memoizee` using for serialisation of values, therefore I assume that they compare references to `contract` until you set `primitive` mode to true and write a serialiser. If contract is created with the new keyword every time, then it will have different reference and the cache wouldn't work. If the `contract` modifies itself every time it's called (eg updates count inside itself) then even serialisation wouldn't help. Therefore, the golden rule is to avoid caching functions that accept non-primitive values.